compiler/rustc_metadata/src/rmeta/table.rs

```diff
@@ -468,7 +467,10 @@ impl<I: Idx, const N: usize, T: FixedSizeEncoding<ByteArray = [u8; N]>> TableBui
 
         let width = self.width;
         for block in &self.blocks {
-            buf.emit_raw_bytes(&block[..width]);
+            buf.write_with(|dest| {
+                *dest = *block;
+                width
+            });
```

Comment:
I tried to explain this with a much larger doc comment on `write_with`. Does the comment help?

Reply:
Yes, I think it does. The doc comment basically says that this is to make sure the width of the write is known (in diff with `write_all` which takes a slice), so this doesn't end up calling memcpy.

Ig those cases could be handled by a `write_array<const N: usize>(&mut self, &[u8; N])`, but it is not necessary to add since we need `write_with` for other cases anyway